### ISSUE TITLE
react/display-name false positive when callback return component

### ISSUE THREAD

**Issue by Akiyamka** [yannickcr/eslint-plugin-react#3104]
```tsx
      <div>
        {statesToComponents({
          loading: <LoadingSpinner />,
          error: (errorMessage) => <ErrorMessage message={errorMessage} />, 
          //     ^^^^^^^^^^^^^^   <- react/display-name error - This is not component, this is callback
          ready: (data) => data.map((item) => <Item key={item.id}  />),
        })}
      </div>
 ```
 
 eslint@7.32.0
 eslint-plugin-react 7.24.0
